MediaBrowser.Controller/Entities/Video.cs

```diff
@@ -550,5 +541,25 @@ public virtual MediaStream GetDefaultVideoStream()
 
             return list;
         }
+
+        private int GetMediaSourceCount(Stack<Guid> callstack)
```

Comment:
I somewhat see how `Stack` might be semantically suitable, but `.Contains` is a linear search. A `HashSet` or similar is a better data structure for this.